src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2724,6 +2724,31 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
         {
             return null;
         }
+
+        public override BoundNode VisitSwitchExpression(BoundSwitchExpression node)
+        {
+            VisitRvalue(node.GoverningExpression);
+            var dispatchState = this.State;
+            var endState = UnreachableState();
+            foreach (var section in node.SwitchSections)
+            {
+                SetState(dispatchState.Clone());
+                VisitPattern(node.GoverningExpression, section.Pattern);
+                SetState(StateWhenTrue);
+                if (section.Guard != null)
+                {
+                    VisitCondition(section.Guard);
+                    SetState(StateWhenTrue);
+                }
+
+                VisitRvalue(section.Value);
+                IntersectWith(ref endState, ref this.State);
```

Comment:
I think the second argument here is passed by `ref` because we call Normalize, which modifies the state. Is it worth trying to move `Normalize` to some central point so we can pass the second argument as `in`? Right now it's not very clear to the reader that the first argument is expected to be mutated, but the second is only expected to mutate as an optimization.

Reply:
I'd be OK with that, but not as part of this PR.

---
In reply to: [166482242](https://github.com/dotnet/roslyn/pull/23357#discussion_r166482242) [](ancestors = 166482242)